### DISCUSSION TITLE
How to validate two arguments in property (is possible the use of two validators?)

### DISCUSSION THREAD

**Discussion by sansanda** [pymeasure/pymeasure#925] [Q&A] [answered]
Hi All,

when you have an instrument property like setting with a command like this "GTXY %d,%d", how can I validate the range of each param?

Many thanks,

David.

**Comment by BenediktBurger** [comment, marked as answer]
the validator has to accept and compare the tuple.
For example, you write `some_property = value_1, value_2`, then the validator will be called with the parameter `value=(value_1, value_2)`.

I think, there is no ready validator for that case, but you could write a validator, which validates both (or more) entries in a tuple according to your needs.

For example:
~~~python
def multi_validator(value, values):
    if not values[0][0] < value[0] < values[0][1]:
        raise ValueError(f"Value 1 {value[0]} is out of bounds")
    if not values[1][0] < value[1] < values[1][1]:
        raise ValueError(f"Value 2 {value[1]} is out of bounds")
    return value


some_property = Instrument.control(...
    values = [[1, 10], [2, 20]],  # first tuple bounds for first entry and second tuple is for second entry.
    validator = multi_validator,
~~~

**Comment by sansanda** [comment]
@bmoneke many thanks. I'll try it.

David
